service/src/main/java/com/codeforcommunity/processor/AuthDatabase.java

```diff
@@ -60,4 +66,37 @@ public boolean isOnBlackList(String signature) {
 
         return count >= 1;
     }
+
+    public void createSecretKey(int userId, String token) throws AuthException {
+        if (!doesUserExist(userId)) {
+            throw new AuthException("User does not exist.");
+        }
+
+        db.insertInto(Tables.VERIFICATION_KEYS).columns(VERIFICATION_KEYS.ID, VERIFICATION_KEYS.USER_ID)
+            .values(token, userId).execute();
+    }
+
+    private boolean doesUserExist(int userId) {
+        Result<NoteUserRecord> userRecord = db.selectFrom(Tables.NOTE_USER).where(NOTE_USER.ID.eq(userId)).fetch();
+        return userRecord.isNotEmpty();
+    }
+
+    public int validateSecretKey(String secretKey) throws AuthException {
+        Timestamp cutoffDate = Timestamp.from(LocalDateTime.now().minusDays(1).atZone(ZoneId.systemDefault()).toInstant());
```

Comment:
You can do this without LocalDateTime. Just use a Timestamp and don't worry about the timezone since we just need a standard epoch time